rotkehlchen/chain/ethereum/modules/ens/decoder.py

```diff
@@ -72,20 +72,27 @@ def _save_hash_mappings_get_fullname(name: str) -> str:
     Saves the namehash -> name and labelhash -> name mappings to the global DB
     cache and gets name with.eth suffix.
 
-    The given name has to be without the .eth suffix and is returned with it
+    The given name has to be without the .eth suffix and is returned with it. In the case that
+    the registered name is invalid we return the name but we don't store it in the database cache.
     """
     full_name = f'{name}.eth'
     with GlobalDBHandler().conn.write_ctx() as write_cursor:
-        globaldb_set_unique_cache_value(
-            write_cursor=write_cursor,
-            key_parts=(CacheType.ENS_NAMEHASH, ENS.namehash(full_name).hex()),
-            value=full_name,
-        )
-        globaldb_set_unique_cache_value(
-            write_cursor=write_cursor,
-            key_parts=(CacheType.ENS_LABELHASH, ENS.labelhash(name).hex()),
-            value=name,
-        )
+        try:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.ENS_NAMEHASH, ens.ENS.namehash(full_name).hex()),
+                value=full_name,
+            )
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.ENS_LABELHASH, ens.ENS.labelhash(name).hex()),
+                value=name,
+            )
+        except ens.exceptions.InvalidName as e:
+            log.error(
+                f'Got an invalid ENS name {name}. namehash and labelhash not stored '
+                f'in the globaldb cache. {e=}',
+            )
```

Comment:
please add a unit test for an invvalid name being passed here and not being saved in the DB